src/com/seafile/seadroid2/transfer/TransferManager.java

```diff
@@ -112,22 +116,28 @@ protected void remove(int taskID) {
             return;
 
         if (!waitingList.isEmpty()) {
```

Comment:
Operations like `isEmpty` should also be synchronized. So we'd better add `synchronized` keyword to the whole method.